bc_obps/registration/migrations/0079_migrate_reg1_operation_statuses.py

```diff
@@ -0,0 +1,120 @@
+# Generated by Django 5.0.11 on 2025-02-19 00:00
+
+from django.db import migrations
+from typing import Dict
+import datetime
+import uuid
+
+"""
+One-time forward-only migration to be applied to prod data.
+Purpose: set the `status` field of Operations (created in Reg1) to match the statuses used in Administration/Registration2 modules.
+Also archives operations with a "Declined" status (only 1 as of Feb 19, 2025)
+
+Reg1 Status        | Reg2 Status
+-------------------+-------------------
+PENDING            | DRAFT
+APPROVED           | DRAFT
+CHANGES_REQUESTED  | DRAFT
+NOT_STARTED        | NOT_STARTED
+DRAFT              | DRAFT
+DECLINED           | Archive the operation
+"""
+
+
+def count_stats(Operation) -> Dict[str, int]:
+    total = Operation.objects.count()
+    declined = Operation.objects.filter(status="Declined").count()
+    approved = Operation.objects.filter(status="Approved").count()
+    not_started = Operation.objects.filter(status="Not Started").count()
+    draft = Operation.objects.filter(status="Draft").count()
+    changes_requested = Operation.objects.filter(status="Changes Requested").count()
+    pending = Operation.objects.filter(status="Pending").count()
+    registered = Operation.objects.filter(status="Registered").count()
+
+    return {
+        'total': total,
+        'declined': declined,
+        'approved': approved,
+        'not_started': not_started,
+        'draft': draft,
+        'changes_requested': changes_requested,
+        'pending': pending,
+        'registered': registered,
+    }
+
+
+def migrate_reg1_operation_statuses(apps, schema_monitor):
+    # import the required Django model
+    Operation = apps.get_model('registration', 'Operation')
+
+    before_stats = count_stats(Operation)
+
+    declined_operations_before = Operation.objects.filter(status="Declined")
+
+    print("\n\n\nDeclined operations - before migration:")
+    for op in declined_operations_before:
+        print(f'{op.id}: archived_by_id={op.archived_by_id}, archived_at={op.archived_at}')
+
+    pending_operations_updated = Operation.objects.filter(status="Pending").update(status="Draft")
+    approved_operations_updated = Operation.objects.filter(status="Approved").update(status="Draft")
+    changes_requested_operations_updated = Operation.objects.filter(status="Changes Requested").update(status="Draft")
+    declined_operations_archived = Operation.objects.filter(status="Declined").update(
+        archived_by_id='c3bc1b69-15de-44ac-b03b-982bf3163406', archived_at=datetime.datetime.now(datetime.timezone.utc)
```

Comment:
Is it OK to have a real uuid in a public repo? (I know we have some in our mock data too, but I think we were trying to avoid where possible.) Possibly we could select it from the User model by some other field?

Reply:
I don't think it's a major security risk but probably not the best idea either. My original thinking was to avoid including her full name in the repo... but it's her github username anyway, so she's basically doxxed herself (as have I). I'll fix this.